sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryAvroUtils.java

```diff
@@ -353,7 +353,7 @@ private static Object convertRequiredField(
       case "BOOL":
       case "BOOLEAN":
         verify(v instanceof Boolean, "Expected Boolean, got %s", v.getClass());
-        return v;
+        return v.toString();
```

Comment:
Storing objects in the JSON class makes the unit tests annoying to write. Generally after round tripping through serialization the objects come back as strings, and tests fail (since .equals() simply compares the objects). This can be tricky to debug since things all format the same.

While this does somewhat increase the in-memory representation of the object, these objects are usually dominated by everything else (e.g. the full schema is stored with every object, and the map contains the full name of every key).